### PR TITLE
Add `eslint-plugin-playwright`

### LINKED ISSUE #2312: Add `eslint-plugin-playwright`
## Problem

In https://github.com/WordPress/openverse/pull/2199 we added `eslint-plugin-jest`. However, this only covers tests run by jest and does not apply any checks to Playwright tests, of which we have many important ones. `eslint-plugin-playwright` has many similar rules but also adds Playwright-specific rules that we could benefit from.

## Description

Add `eslint-plugin-playwright` and apply it to `frontend/test/playwright` and `frontend/test/storybook` using `overrides`, similar to how `eslint-plugin-jest` is applied only to the `frontend/test/unit` directory.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #2312 by @sarayourfriend 

## Description

See #2571 for additional details regarding the disabled `expect-expect` rule.

This PR adds the plugin and resolves the various warnings and errors it produced. Where possible I preferred to remove conditionals rather than merely silencing the warnings, so this changed the approach for handling language direction in the storybook tests. I did need to selectively disable some rules, but I tried to add explanatory comments for those.

## Testing Instructions

All Playwright tests should continue to pass as they did before. Checkout this branch and play around with the new eslint run (run `pnpm install` first) and confirm it works as expected.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by obulat — CHANGES_REQUESTED** [review]
Thank you for adding eslint to playwright! Nice to see some things cleaned up.

I don't particularly like the `resolves.toBeUndefined` pattern, and would prefer to convert it to better `expects`. 
I tried to refactor the checkbox assertion function locally to make it clearer. There are lots of changes, so I'm attaching them as a patch here. Do you think it would be better to update the checkbox assertions like this?
[Use_expect_instead_of_`_resolves`.patch](https://github.com/WordPress/openverse/files/11968433/Use_expect_instead_of_._resolves.patch)

**Comment by sarayourfriend** [comment]
I didn't like it either but wanted to minimise the changes.

I like your approach, except that I think it would be better to just use `expect(getCheckbox(...)).isChecked()`, just so the test reads cleaner and to follow the "web-first assertions" idea.

I'll apply the patch here, thanks for sharing that!

File: frontend/test/playwright/e2e/external-sources.spec.ts [hunk 1]
```diff
@@ -25,10 +25,6 @@ test("sends correct analytics events", async ({ page, context }) => {
     (event) => event.n === "SELECT_EXTERNAL_SOURCE"
   )
 
-  if (!viewEvent || !selectEvent) {
```
Comment (AetherUnbound) [inline frontend/test/playwright/e2e/external-sources.spec.ts:28]: Why was this removed?

**Review by AetherUnbound — APPROVED** [review]
LGTM! Just one question but its answer is likely related to my unfamiliarity with testing Javascript 🙂

File: frontend/test/playwright/e2e/external-sources.spec.ts (hunk 1 above)
Comment (sarayourfriend) [reply frontend/test/playwright/e2e/external-sources.spec.ts:28]: It duplicates the falsy check in the assert event payload utility.

**Comment by sarayourfriend** [comment]
Unfortunately the now failing tests are coming from tests modified by the patch I've applied from @obulat. I'm trying to discern why they are failing. For the filters.spec tests the checkboxes fail `toBeVisible` and I suspect there's something wrong with the selection method, which may mean that tests using `getCheckbox` generally might not be stable. I'll keep digging and push a fix soon, I hope.

**Comment by sarayourfriend** [comment]
Hmm, the storybook test is flaky. I will open an issue for it tomorrow (I need to get off the computer for the day).

File: frontend/test/playwright/e2e/filters.spec.ts [hunk 2]
```diff
@@ -120,23 +123,64 @@ breakpoints.describeMobileAndDesktop(() => {
...
+    // Enable the commercial use filter
+    await page.locator('label:has-text("Use commercially")').click()
+
+    await expect(
+      page.getByRole("checkbox", { name: "Use commercially" })
+    ).toBeChecked()
+
+    // Because we checked "Use commercially", licenses that disallow commercial
+    // use will be disabled and the rest will still be enabled.
+    // Additionally, none of the checkboxes will be checked becuase we've only
```
Comment (obulat) [inline frontend/test/playwright/e2e/filters.spec.ts:166]:
```suggestion
    // Additionally, none of the checkboxes will be checked because we've only
```

**Review by obulat — APPROVED** [review]
> Unfortunately the now failing tests are coming from tests modified by the patch I've applied from @obulat. I'm trying to discern why they are failing. For the filters.spec tests the checkboxes fail toBeVisible and I suspect there's something wrong with the selection method, which may mean that tests using getCheckbox generally might not be stable. I'll keep digging and push a fix soon, I hope.

I guess there is not much use for the `getCheckbox` now, the conversion makes it easier to read.

**Comment by obulat** [comment]
> Hmm, the storybook test is flaky. I will open an issue for it tomorrow (I need to get off the computer for the day).
> 
> Failed run: https://github.com/WordPress/openverse/actions/runs/5516590713/jobs/10058165697?pr=2574
> 
> Test artefact: [storybook_test_results.zip](https://github.com/WordPress/openverse/files/12011954/storybook_test_results.zip)

This seems to be a rendering problem with Linux in Docker. The fonts are pixellated in the updated snapshot. Adding retries could be the best solution.

File: frontend/test/playwright/e2e/external-sources.spec.ts (hunk 1 above)
Comment (AetherUnbound) [reply frontend/test/playwright/e2e/external-sources.spec.ts:28]: Got it, thank you!
